homeassistant/components/matter/select.py

```diff
@@ -0,0 +1,186 @@
+"""Matter ModeSelect Cluster Support."""
+
+from __future__ import annotations
+
+from chip.clusters import Objects as clusters
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory, Platform
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .entity import MatterEntity
+from .helpers import get_matter
+from .models import MatterDiscoverySchema
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Matter ModeSelect from Config Entry."""
+    matter = get_matter(hass)
+    matter.register_platform_handler(Platform.SELECT, async_add_entities)
+
+
+class MatterModeSelectEntity(MatterEntity, SelectEntity):
+    """Representation of a select entity from Matter (Mode) Cluster attribute(s)."""
+
+    async def async_select_option(self, option: str) -> None:
+        """Change the selected mode."""
+        cluster: clusters.ModeSelect = self._endpoint.get_cluster(
+            self._entity_info.primary_attribute.cluster_id
+        )
+        # select the mode ID from the label string
+        for mode in cluster.supportedModes:
+            if mode.label != option:
+                continue
+            await self.matter_client.send_device_command(
+                node_id=self._endpoint.node.node_id,
+                endpoint_id=self._endpoint.endpoint_id,
+                command=cluster.Commands.ChangeToMode(newMode=mode.mode),
+            )
+            break
+
+    @callback
+    def _update_from_device(self) -> None:
+        """Update from device."""
+        # NOTE: cluster can be ModeSelect or a variant of that,
+        # such as DishwasherMode. They all have the same characteristics.
+        cluster: clusters.ModeSelect = self._endpoint.get_cluster(
```

Comment:
If you want to guarantee that they all work the same, you could just define all types here and type checking will verify they remain all the same.

Reply:
yeah but that becomes an awful long list

Reply:
ok maybe it was the setting of the moon yesterday but today it works. Weird.
So adjusted it with a type alias